core/src/env2/env_access/immutable.rs

```diff
@@ -38,11 +52,87 @@ use core::cell::RefCell;
 ///
 /// Using `EnvAccessMut` is preferable since it performs these access checks at
 /// compile-time.
-pub struct EnvAccess<T> {
+pub struct EnvAccess<E> {
     /// Allows accessing the inner environment by `&self` instead of `&mut self`.
     ///
     /// This is important to make `DynEnv` work also in conjunction with `&self` messages.
-    access: RefCell<EnvAccessMut<T>>,
+    pub(crate) access: RefCell<EnvAccessMut<E>>,
+}
+
+impl<E> core::fmt::Debug for EnvAccess<E> {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        f.debug_struct("EnvAccess").finish()
+    }
+}
+
+#[cfg(feature = "ink-generate-abi")]
+impl<E> type_metadata::HasTypeId for EnvAccess<E>
+where
+    E: type_metadata::Metadata,
+{
+    fn type_id() -> type_metadata::TypeId {
+        type_metadata::TypeIdCustom::new(
+            "EnvAccess",
+            type_metadata::Namespace::from_module_path(module_path!())
```

Comment:
When executing `RUSTUP_TOOLCHAIN=nightly scripts/check-*` I run into some errors. These two are among them:
```
error[E0599]: no function or associated item named `from_module_path` found for type `type_metadata::type_id::Namespace<_>` in the current scope               
  --> /home/michi/projects/ink/core/src/env2/env_access/immutable.rs:76:39                                                                                     
   |                                                                                                                                                           
76 |             type_metadata::Namespace::from_module_path(module_path!())                                                                                    
   |                                       ^^^^^^^^^^^^^^^^ function or associated item not found in `type_metadata::type_id::Namespace<_>`                    
                                                                                                                                                               
error[E0599]: no function or associated item named `from_module_path` found for type `type_metadata::type_id::Namespace<_>` in the current scope               
  --> /home/michi/projects/ink/core/src/env2/test/accessor.rs:98:39                                                                                            
   |                                                                                                                                                           
98 |             type_metadata::Namespace::from_module_path(module_path!())                                                                                    
   |                                       ^^^^^^^^^^^^^^^^ function or associated item not found in `type_metadata::type_id::Namespace<_>`                    
                                                                                                                                                               
error: aborting due to 2 previous errors
```

Reply:
Try to run `cargo update` because this is depending on some "more recent" changes in the `type-metadata` crate.

Reply:
Thanks, it runs fine now!